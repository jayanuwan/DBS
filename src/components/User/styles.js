import styled from 'styled-components';

export const UserItemContainer = styled.div`
	width: 100%;
	min-height: 200px;
`;

export const UserItem = styled.div`
	width: 100%;
	padding: 10px;
`;
