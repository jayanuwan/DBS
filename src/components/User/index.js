import React from 'react';
import * as SC from './styles';
import { Column } from '../styles';

const User = ({ name, email, phone, website }) => {
	return (
		<Column xs="12" sm="6" md="4" data-testid="user-details">
			<SC.UserItemContainer>
				<SC.UserItem>{name}</SC.UserItem>
				<SC.UserItem>
					<a href={'mailto:' + email}>{email}</a>
				</SC.UserItem>
				<SC.UserItem>
					<a href={'tel:' + phone}>{phone}</a>
				</SC.UserItem>
				<SC.UserItem>{website}</SC.UserItem>
			</SC.UserItemContainer>
		</Column>
	);
};

export default User;
