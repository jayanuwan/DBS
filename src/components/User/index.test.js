import React from 'react';
import { getByTestId, render } from '@testing-library/react';
import User from './index';

const data = {
	name: '',
	phone: '',
	email: '',
	web: '',
};
/**
 * @jest-environment jsdom
 */
describe('user details', () => {
	it('renders user Details', () => {
		const { getByTestId } = render(
			<User
				name={data.name}
				email={data.email}
				phone={data.phone}
				website={data.web}
			/>
		);

		expect(getByTestId('user-details')).toMatchSnapshot();
	});
});
