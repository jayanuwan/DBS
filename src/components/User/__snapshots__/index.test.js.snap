// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user details renders user Details 1`] = `
<div
  class="sc-dlnjwi ZrTjy"
  data-testid="user-details"
>
  <div
    class="sc-bdnxRM AFMeB"
  >
    <div
      class="sc-gtsrHT dJJWhK"
    />
    <div
      class="sc-gtsrHT dJJWhK"
    >
      <a
        href="mailto:"
      />
    </div>
    <div
      class="sc-gtsrHT dJJWhK"
    >
      <a
        href="tel:"
      />
    </div>
    <div
      class="sc-gtsrHT dJJWhK"
    />
  </div>
</div>
`;
